digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/SpeedLimitFiller.scala

```diff
@@ -93,6 +93,18 @@ object SpeedLimitFiller {
     }
   }
 
+  private def droppedSegmentWrongDirection(roadLink: RoadLink, segments: Seq[SpeedLimit], changeSet: ChangeSet): (Seq[SpeedLimit], ChangeSet) = {
+    if (roadLink.trafficDirection == TrafficDirection.BothDirections) {
+      (segments, changeSet)
+    } else {
+      val droppedAssetIds = (roadLink.trafficDirection match {
+        case TrafficDirection.TowardsDigitizing => segments.filter(s => s.sideCode == SideCode.AgainstDigitizing)
+        case _ => segments.filter(s => s.sideCode == SideCode.TowardsDigitizing)
+      }).map(_.id)
+
+      (segments.filterNot(s => droppedAssetIds.contains(s.id)), changeSet.copy(droppedAssetIds = changeSet.droppedAssetIds++ droppedAssetIds))
+    }
```

Comment:
droppedAssetIds should be change to expiredAssetTds. Drop isn't exist on SpeedLimits